src/qililab/qprogram/qprogram.py

```diff
@@ -316,6 +335,7 @@ def set_gain(self, bus: str, gain: float):
         """
         operation = SetGain(bus=bus, gain=gain)
         self._active_block.append(operation)
+        self._buses.add(bus)
 
     def set_offset(self, bus: str, offset_path0: float, offset_path1: float):
```

Comment:
Shouldn't this function need also the `@requires_domain("gain", Domain.Voltage)`?

Reply:
Yes, you right. Nice catch. I will add it.

Reply:
Done